### PR TITLE
fix: make v2 with modules run

### PR DESCRIPTION
**Pull request by srindom**

### CODE REVIEW

File: packages/medusa/src/loaders/index.ts [hunk 1]
```diff
@@ -91,12 +93,10 @@ async function loadMedusaV2({ directory, expressApp }) {
     next()
   })
 
-  const featureFlagRouter = featureFlagsLoader(configModule, Logger)
-
   const pgConnection = await pgConnectionLoader({ container, configModule })
 
   container.register({
-    [ContainerRegistrationKeys.MANAGER]: asValue(dataSource.manager),
+    // [ContainerRegistrationKeys.MANAGER]: asValue(dataSource.manager),
```
Comment (srindom) [inline packages/medusa/src/loaders/index.ts]: Is this safe to do? cc: @adrien2p

File: packages/medusa/src/commands/migrate.js [hunk 2]
```diff
@@ -70,15 +73,32 @@ const main = async function ({ directory }) {
...
-    await runIsolatedModulesMigration(configModule)
-
-    await runLinkMigrations(directory)
+    if (featureFlagRouter.isFeatureEnabled(MedusaV2Flag.key)) {
+      const container = createMedusaContainer()
+      const pgConnection = await pgConnectionLoader({ configModule, container })
+      container.register({
+        [ContainerRegistrationKeys.CONFIG_MODULE]: asValue(configModule),
+        [ContainerRegistrationKeys.LOGGER]: asValue(Logger),
+        [ContainerRegistrationKeys.MANAGER]: asValue(dataSource.manager),
```
Comment (srindom) [inline packages/medusa/src/commands/migrate.js]: can we get rid of this? cc: @adrien2p?

File: packages/medusa/src/loaders/index.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/index.ts]: Yes i believe so, modules does not rely on the manager

File: packages/medusa/src/commands/migrate.js (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/commands/migrate.js]: You can just keep the logger

File: packages/medusa/package.json [hunk 3]
```diff
@@ -70,7 +70,7 @@
     "core-js": "^3.6.5",
     "cors": "^2.8.5",
     "cross-spawn": "^7.0.3",
-    "dotenv": "^16.0.3",
+    "dotenv": "^16.4.5",
```
Comment (adrien2p) [inline packages/medusa/package.json:73]: In the future with node 21.7 it will not be needed anymore

File: packages/utils/src/common/deep-copy.ts [hunk 4]
```diff
@@ -10,10 +10,6 @@ import { isObject } from "./is-object"
 export function deepCopy<T extends Record<any, any> = Record<any, any>>(
   obj: T | T[]
 ): T | T[] {
-  if (typeof structuredClone != "undefined") {
```
Comment (adrien2p) [inline packages/utils/src/common/deep-copy.ts:13]: Nit could you update the tsdoc please

**Review by adrien2p — APPROVED** [review]

**Review by adrien2p — APPROVED** [review]

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Pull request merged**